platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReaderPbj.java

```diff
@@ -0,0 +1,465 @@
+        return new DataFileIteratorPbj(path, metadata, dataItemSerializer);
+    }
+
+    /**
+     * Read data item bytes from file at dataLocation and deserialize them into the Java object, if
+     * requested.
+     *
+     * @param dataLocation The file index combined with the offset for the starting block of the
+     *     data in the file
+     * @return Deserialized data item, or {@code null} if deserialization is not requested
+     * @throws IOException If there was a problem reading from data file
+     * @throws ClosedChannelException if the data file was closed
+     */
+    @Override
+    public D readDataItem(final long dataLocation) throws IOException {
+        final long byteOffset = DataFileCommon.byteOffsetFromDataLocation(dataLocation);
+        final BufferedData data = read(byteOffset);
+        return dataItemSerializer.deserialize(data);
+    }
+
+    @Override
+    public Object readDataItemBytes(final long dataLocation) throws IOException {
+        final long byteOffset = DataFileCommon.byteOffsetFromDataLocation(dataLocation);
+        return read(byteOffset);
+    }
+
+    /**
+     * Get the size of this file in bytes. This method should only be called for files available to
+     * merging (compaction), i.e. after they are fully written.
+     *
+     * @return file size in bytes
+     */
+    public long getSize() {
+        return fileSizeBytes.get();
+    }
+
+    /** Equals for use when comparing in collections, based on matching file paths */
+    @SuppressWarnings("rawtypes")
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final DataFileReader that = (DataFileReader) o;
+        return path.equals(that.getPath());
+    }
+
+    /** hashCode for use when comparing in collections, based on file path */
+    @Override
+    public int hashCode() {
+        return path.hashCode();
+    }
+
+    /** Compares this Data File to another based on creation date and index */
+    @Override
+    public int compareTo(final DataFileReader o) {
+        Objects.requireNonNull(o);
+        if (this == o) {
+            return 0;
+        }
+        final int res = metadata.getCreationDate().compareTo(o.getMetadata().getCreationDate());
+        return (res != 0)
+                ? res
+                : Integer.compare(metadata.getIndex(), o.getMetadata().getIndex());
+    }
+
+    /** ToString for debugging */
+    @Override
+    public String toString() {
+        return Integer.toString(metadata.getIndex());
+    }
+
+    /**
+     * Get if the DataFile is open for reading.
+     *
+     * @return True if file is open for reading
+     */
+    public boolean isOpen() {
+        return open.get();
+    }
+
+    /** Close this data file, it can not be used once closed. */
+    public void close() throws IOException {
```

Comment:
Probably should add the `@Override` annotation here.

Reply:
Added `@Override` to all methods in this class and in `DataFileReaderJdb` and removed JavaDocs (since they are in the the interface)